platform/linux-generic/pktio/dpdk.c

```diff
@@ -301,6 +301,48 @@ static struct rte_mempool_ops ops_stack = {
 
 MEMPOOL_REGISTER_OPS(ops_stack);
 
+#define HAS_IP4_PROTO(m, proto) ((m->packet_type & RTE_PTYPE_L3_MASK) == proto)
+#define HAS_IP4_PROTOS(m) (HAS_IP4_PROTO(m, RTE_PTYPE_L3_IPV4) ||\
+			   HAS_IP4_PROTO(m, RTE_PTYPE_L3_IPV4_EXT) ||\
+			   HAS_IP4_PROTO(m, RTE_PTYPE_L3_IPV4_EXT_UNKNOWN))
```

Comment:
This looks unnecessarily complex. You can simply do e.g.
`#define PTYPE_IPV4 (RTE_PTYPE_L3_IPV4 | RTE_PTYPE_L3_IPV4_EXT | RTE_PTYPE_L3_IPV4_EXT_UNKNOWN)`
and OR that with `(m->packet_type & RTE_PTYPE_L3_MASK)`.

Reply:
I am not sure what you mean:

PTYPE_IPV4 = (RTE_PTYPE_L3_IPV4 | RTE_PTYPE_L3_IPV4_EXT | RTE_PTYPE_L3_IPV4_EXT_UNKNOWN) = 0001 0000 | 0011 0000 | 1001 0000 = 1011 0000

If packet is IPv6 with extensions:

m->packet_type & RTE_PTYPE_L3_MASK = RTE_PTYPE_L3_IPV6_EXT = 1100 0000

1011 0000 | 1100 0000 is not helping us.

Reply:
True, my mistake. Looking at the binary presentations it looks like you can simply use RTE_PTYPE_L3_IPV4 and RTE_PTYPE_L3_IPV6 values and do OR operation. Those bits won't overlap.

Reply:
True, yet you cannot really trust that those defines will keep their values in the next dpdk versions. This is how APIs are extended.
We should not design this implementation based on the values but based on defines (RTE_PTYPE_L3_XXX and RTE_PTYPE_L3_MASK).